lib/hanami/environment.rb

```diff
@@ -461,7 +459,7 @@ def set_hanami_env_vars!
     # @since 0.2.0
     # @api private
     def set_application_env_vars!
-      Dotenv.load     root.join(DEFAULT_DOTENV)
+      return unless defined?(Dotenv)
       Dotenv.overload root.join(DEFAULT_DOTENV_ENV % environment)
```

Comment:
@jodosha 
`#load` and `#overload` has only one difference `#load` uses `#apply` and `#overload` uses `#apply!` [proof link](https://github.com/bkeepers/dotenv/blob/c44af19abeb12240995358fb350c1dab42d3c0fe/lib/dotenv/environment.rb#L20-L26)

```
    def apply
      each { |k, v| ENV[k] ||= v }
    end

    def apply!
      each { |k, v| ENV[k] = v }
    end
```

So have to use `#overload` because if the variable exists it should be overrided.